### ISSUE TITLE
str not recognized as Iterable[Literal[T]]

### ISSUE THREAD

**Issue by johanvergeer** [python/mypy#10339]
**Bug Report**

I have a function taking in an `Iterable` of movements, which are defined as `Literal` strings.

```python
def move(command: Iterable[Literal["f", "b", "l", "r"]]) -> None:
    for c in command:
        print(c)
```

But an error is thrown when I call this function with a `str` only containing the allowed characters.

```python
move("ffbblr")
```

```bash
$ mypy iter.py
iter.py:9: error: Argument 1 to "move" has incompatible type "str"; expected "Iterable[Union[Literal['f'], Literal['b'], Literal['l'], Literal['r']]]"
iter.py:9: note: Following member(s) of "str" have conflicts:
iter.py:9: note:     Expected:
iter.py:9: note:         def __iter__(self) -> Iterator[Union[Literal['f'], Literal['b'], Literal['l'], Literal['r']]]
iter.py:9: note:     Got:
iter.py:9: note:         def __iter__(self) -> Iterator[str]
```

Everything is fine when I call the same function with a list of `str`:

```python
move(["f", "f", "b", "b", "l", "r"])
```

```bash
$ mypy iter.py
Success: no issues found in 1 source file
```

**To Reproduce**

1. Copy the function above
2. Copy the call to the function above with the string
3. Run the script

**Expected Behavior**

Mypy should be able to handle this, since a `str` implements the `Iterable` protocol.

**Actual Behavior**

See the errors above

**Your Environment**

- Mypy version used: 0.812
- Mypy command-line flags: 

```bash
$ mypy iter.py
```

- Mypy configuration options from `mypy.ini` (and other config files):

```ini
[mypy]
python_version = 3.9
warn_no_return = True
disallow_untyped_defs=True
```

- Python version used: 3.9
- Operating system and version: MacOS Cataline 10.15.7

**Comment by JadeKim042386** [comment]
Generally, literals are a notation for representing a fixed value in source code. Therefore, I think the above error occured because string changes as it it implemented into and iterable data structure.
Or this may be due to changed in version 3.9.1 described in [typing.Literal](https://docs.python.org/3/library/typing.html#typing.Literal).

**Comment by hauntsaninja** [comment]
This is expected given the typeshed stub for `str.__iter__`. I'm not sure how mypy could type check this without a lot of special case understanding of `str.__iter__`, and I don't think this use case is important enough to merit that.
